core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -394,24 +395,48 @@ private RelationPlan createTableCreationPlan(Analysis analysis, Query query)
         }
 
         ConnectorTableMetadata tableMetadata = create.getMetadata().orElseThrow();
-
         Optional<TableLayout> newTableLayout = create.getLayout();
 
-        List<String> columnNames = tableMetadata.getColumns().stream()
-                .filter(column -> !column.isHidden()) // todo this filter is redundant
-                .map(ColumnMetadata::getName)
-                .collect(toImmutableList());
+        List<Symbol> visibleFieldMappings = getVisibleFieldMappings(plan);
 
         String catalogName = destination.getCatalogName();
         CatalogHandle catalogHandle = metadata.getCatalogHandle(session, catalogName)
                 .orElseThrow(() -> semanticException(CATALOG_NOT_FOUND, query, "Destination catalog '%s' does not exist", catalogName));
+
+        Assignments.Builder assignments = Assignments.builder();
+        ImmutableList.Builder<ColumnMetadata> finalColumnsBuilder = ImmutableList.builder();
+
+        forEachPair(tableMetadata.getColumns().stream(), visibleFieldMappings.stream(), (column, fieldMapping) -> {
+            Expression expression;
+            Type tableType = column.getType();
+            Type queryType = symbolAllocator.getTypes().get(fieldMapping);
+            if (queryType.equals(tableType) || typeCoercion.isTypeOnlyCoercion(queryType, tableType)) {
+                expression = fieldMapping.toSymbolReference();
```

Comment:
It's not OK to omit type-only coercions in the plan.
The PlanNodes are strictly typed and all types must match.

Reply:
The same code is there for `INSERT`. isn't the point of coercion that the types match?